packages/core/src/App/Containers/RealAccountSignup/account-wizard.jsx

```diff
@@ -60,6 +63,28 @@ const AccountWizard = props => {
     const [should_accept_financial_risk, setShouldAcceptFinancialRisk] = React.useState(false);
     const is_high_risk_client_for_mt5 = useIsClientHighRiskForMT5();
 
+    const analytic_data = React.useMemo(
+        () => ({
+            form_source: document.referrer,
+            form_name: 'real_account_signup_form',
+            landing_company: props.real_account_signup_target,
+        }),
+        [props.real_account_signup_target]
+    );
+
+    const trackEvent = React.useCallback(
+        payload => {
+            if (props.real_account_signup_target === 'maltainvest') return;
+
+            Analytics.trackEvent('ce_real_account_signup_form', {
+                current_step: STEP_IDENTIFIERS[payload.step_num],
+                ...analytic_data,
+                ...payload,
+            });
+        },
+        [analytic_data, props.real_account_signup_target]
+    );
```

Comment:
it's brilliant idea to use useMemo for object to increase performance a bit 👍 
but useCallback here not necessary and will decrease what we won by useMemo

My suggestion is to use Analytics.trackEvent directly in the places where we have to track smth
Why I think so? Analytics package already provide only 1 function in memory and this function will never re-initialise due to the react re-render how it's happen for common function

`props.real_account_signup_target !== 'maltainvest' && Analytics.trackEvent('ce_real_account_signup_form', {...})`

Reply:
Thanks @NikitK-deriv for the input. I've moved analytics_data to the function itself as it made sense as suggested by @jim-deriv 

I created the `trackEvent` so I won't have to add `real_account_signup_target` condition everywhere I add the Analytics. I prefer it like however i can remove the useCallback if you suggest